examples/example-xds/src/main/java/io/grpc/examples/helloworldxds/HelloWorldClientXds.java

```diff
@@ -0,0 +1,105 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.helloworldxds;
+
+import io.grpc.ChannelCredentials;
+import io.grpc.Grpc;
+import io.grpc.InsecureChannelCredentials;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.StatusRuntimeException;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.NettyChannelBuilder;
+import io.grpc.xds.XdsChannelCredentials;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.net.ssl.SSLException;
+
+/**
+ * A simple xDS client that requests a greeting from the {@link HelloWorldServerXds}.
+ */
+public class HelloWorldClientXds {
+  private static final Logger logger = Logger.getLogger(HelloWorldClientXds.class.getName());
+  private final ManagedChannel channel;
+  private final GreeterGrpc.GreeterBlockingStub blockingStub;
+
+  /** Construct client connecting to HelloWorld server at {@code host:port}. */
+  public HelloWorldClientXds(String target, boolean useXdsCreds) throws SSLException {
+    this.channel =
+        Grpc.newChannelBuilder(
+                target,
+                useXdsCreds
+                    ? XdsChannelCredentials.create(InsecureChannelCredentials.create())
+                    : InsecureChannelCredentials.create())
+            .build();
+    blockingStub = GreeterGrpc.newBlockingStub(this.channel);
+  }
+
+  public void shutdown() throws InterruptedException {
+    channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+  }
+
+  /** Say hello to server. */
+  public void greet(String name) {
+    logger.info("Will try to greet " + name + " ...");
+    HelloRequest request = HelloRequest.newBuilder().setName(name).build();
+    HelloReply response;
+    try {
+      response = blockingStub.sayHello(request);
+    } catch (StatusRuntimeException e) {
+      logger.log(Level.WARNING, "RPC failed: {0}", e.getStatus());
+      return;
+    }
+    logger.info("Greeting: " + response.getMessage());
+  }
+
+  /**
+   * Greet server. If provided, the second element of {@code args} is the name to use in the
+   * greeting.
+   */
+  public static void main(String[] args) throws Exception {
+    String user = "xds-client";
+    boolean useXdsCreds = false;
+    if (args.length < 1 || args.length > 3) {
+      System.out.println("USAGE: HelloWorldClientXds target [name [--secure]]\n");
```

Comment:
:-/ You changed the order compared to the helloworld example. I can understand why to a degree, but it would probably be better to just have both names required. This will also impact documentation.

Reply:
`target` should be required and name can be optional so this order is better than the one in helloworld. But I guess we cannot change helloworld so I'll make both of them required.

Reply:
I'd be fine if it behaved identically to before: it defaults to not using xds but when you specify an xds string it kicks into action.